proposals/drafts/campaign_finance_filings.rst

```diff
@@ -0,0 +1,298 @@
+    the basic process of filing a document with an agency.
+
+Rationale
+=========
+
+Political committees file statements with their regulators in order to disclose
+funders, expenditures, organizational ties and other financial details.
+Filings encapsulate some number of possible specific disclosures, and some
+filings can contain more than one type of disclosure, so this proposed option is
+to consider each actual filing as an object and then each specific disclosed
+piece of information as associated with that object. So a CommitteeNameChange
+would be associated with a Filing, along with a CommitteeAddressChange and a
+CommitteeOfficerAddition. A series of CommitteeContributions and
+CommitteeExpenses would be attached to one Filing. This way we don't need to
+model the specific disclosure forms in each state, and just directly extract
+the most meaningful parts of the filing.
+
+We want to make it easy for people to do manipulations on top of this data,
+while ensuring that this system itself does as little manipulation as possible
+to the underlying data from reports filed. For instance, many records include
+Person entities - these are deterministically extracted from just the available
+filing data, e.g. an address, employer, occupation and name are all extracted
+for each contribution. Deduplication can then happen with each of these
+entities, but they are themselves totally reproducible from just the filing
+information.
+
+This model entails that Filings have Sections - a Filing can, and often will,
+have multiple Sections. As much of the specific logic as possible should be
+pushed down into Sections, which offer flexibility by allowing us to compose
+them together to reproduce the data from state-specific reports. We can model
+a particular state filing form as a Filing with a specific set of expected
+Sections to be present. Some attributes that might make sense as properties of a
+Filing, such as Amendments, are here presented as Sections; the line between the
+two is muddy, but this proposal tries to keep Filings as minimal as possible.
+
+Finally, the different entity types presented here present either current or
+likely future conflicts with other entities in the Open Civic Data namespace, so
+this proposal doesn't make any attempt to resolve those conflicts since they
+seem ancillary for now.
+
+TODO
+----
+* Make filing activity actions use same pattern as OCD bill actions (@aepton)
+
+Questions to answer
+-------------------
+* How to model Committee Status and Committee Purpose inside of OCD Organization
+  models?
+* How to reconcile multiple reports that describe the same contribution,
+  expenditure or other event?
+* How to handle different types of Committees beyond just imputing their types
+  based on (say) whether they are oriented toward more than one Candidate?
+
+Implementation
+==============
+
+Filing
+------
+
+id
+    Open Civic Data-style id in the format ``ocd-cf-filing/{{uuid}}``
```

Comment:
@fgregg ok, cool - added an original_id optional field to Filing (and a few other objects) to capture the state-generated ID in cases where we care about it, and otherwise I'm happy to assume the IDs of every object are not human-readable, beyond being namespaced.

@jsfenfen My current proposal is to have a field called invalidates_prior_versions that allows us to have amendment actions on filings which either wipe out everything previously disclosed, or don't. I'm thinking amendments should be handled using a combination of that field (if we even need it) and lists of all transactions, etc contained in that filing.

So if you have filing A with transactions B, C and D; and then you have amendment E with transactions B and D; then you've got a lot of duplication but you can also just look at the most recent filing to see all the current versions of the currently-disclosed transactions.